server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -839,58 +842,90 @@ export function configureWebSocketServices(
+						getLumberBaseProperties(connection.documentId, connection.tenantId),
+					);
+					// eslint-disable-next-line @typescript-eslint/no-floating-promises
+					connection.disconnect().catch((error) => {
+						const errorMsg = `Failed to disconnect client ${clientId} from orderer connection.`;
+						Lumberjack.error(
+							errorMsg,
+							getLumberBaseProperties(connection.documentId, connection.tenantId),
+							error,
 						);
+					});
+					if (isClientConnectivityCountingEnabled && throttleAndUsageStorageManager) {
+						const connectionTimestamp = connectionTimeMap.get(clientId);
+						if (connectionTimestamp) {
+							removeAndStoreP.push(
+								storeClientConnectivityTime(
+									clientId,
+									connection.documentId,
+									connection.tenantId,
+									connectionTimestamp,
+									throttleAndUsageStorageManager,
+								),
+							);
+						}
 					}
 				}
-			}
-			// Send notification messages for all client IDs in the room map
-			for (const [clientId, room] of roomMap) {
-				const messageMetaData = getMessageMetadata(room.documentId, room.tenantId);
-				// excluding summarizer for total client count.
-				if (connectionTimeMap.has(clientId)) {
-					connectionCountLogger.decrementConnectionCount();
+				// Send notification messages for all client IDs in the room map
+				for (const [clientId, room] of roomMap) {
+					const messageMetaData = getMessageMetadata(room.documentId, room.tenantId);
+					// excluding summarizer for total client count.
+					if (connectionTimeMap.has(clientId)) {
+						connectionCountLogger.decrementConnectionCount();
+					}
+					logger.info(`Disconnect of ${clientId} from room`, { messageMetaData });
+					Lumberjack.info(
+						`Disconnect of ${clientId} from room`,
+						getLumberBaseProperties(room.documentId, room.tenantId),
+					);
+					removeAndStoreP.push(
+						clientManager.removeClient(room.tenantId, room.documentId, clientId),
+					);
+					socket.emitToRoom(getRoomId(room), "signal", createRoomLeaveMessage(clientId)).catch((error) => {
+						const errorMsg = `Failed to emit signal to room ${clientId}, ${getRoomId(room)}.`;
+						Lumberjack.error(
+							errorMsg,
+							getLumberBaseProperties(room.documentId, room.tenantId),
+							error,
+						);
+					});
 				}
-				logger.info(`Disconnect of ${clientId} from room`, { messageMetaData });
-				Lumberjack.info(
-					`Disconnect of ${clientId} from room`,
-					getLumberBaseProperties(room.documentId, room.tenantId),
-				);
-				removeAndStoreP.push(
-					clientManager.removeClient(room.tenantId, room.documentId, clientId),
-				);
-				socket.emitToRoom(getRoomId(room), "signal", createRoomLeaveMessage(clientId));
+				// Clear socket tracker upon disconnection
+				if (socketTracker) {
+					socketTracker.removeSocket(socket.id);
+				}
+				await Promise.all(removeAndStoreP);
+				if (connectionsMap.size >= 1) {
+					const connections = Array.from(connectionsMap.values());
```

Comment:
To get clientCount from redis, should we use roomMap instead? I think connectionMap does not include read clients. So if there are only readers, it could be inaccurate?